src/screens/AppContainer/AppContainer.tsx

```diff
@@ -1,8 +1,19 @@
+import { useEffect } from 'react';
 import { Text, SafeAreaView } from 'react-native';
+import { useNetInfo } from '@react-native-community/netinfo';
 import PlatformStatusBar from '@components/platformStatusBar';
 import globalStyles from '@themes/globalStyles';
 
 const AppContainer = () => {
+  const netInfo = useNetInfo();
+
+  useEffect(() => {
+    if (netInfo.isConnected !== null) {
+      const { isConnected } = netInfo;
```

Comment:
don't destructure it in 12th line
as you are also using same value in 11th line, do like this-

useEffect(() => {
const { isConnected } = netInfo;
    if (isConnected !== null) {
      if (!isConnected) throw new Error('No internet connection');
    }
  }, [netInfo]);